heroes.json

```diff
@@ -0,0 +1 @@
],"legs":2},{"id":100,"name":"npc_dota_hero_tusk","localized_name":"Tusk","primary_attr":"str","attack_type":"Melee","roles":["Initiator","Disabler","Nuker"],"legs":2},{"id":101,"name":"npc_dota_hero_skywrath_mage","localized_name":"Skywrath Mage","primary_attr":"int","attack_type":"Ranged","roles":["Support","Nuker","Disabler"],"legs":2},{"id":102,"name":"npc_dota_hero_abaddon","localized_name":"Abaddon","primary_attr":"str","attack_type":"Melee","roles":["Support","Carry","Durable"],"legs":2},{"id":103,"name":"npc_dota_hero_elder_titan","localized_name":"Elder Titan","primary_attr":"str","attack_type":"Melee","roles":["Initiator","Disabler","Nuker","Durable"],"legs":2},{"id":104,"name":"npc_dota_hero_legion_commander","localized_name":"Legion Commander","primary_attr":"str","attack_type":"Melee","roles":["Carry","Disabler","Initiator","Durable","Nuker"],"legs":2},{"id":105,"name":"npc_dota_hero_techies","localized_name":"Techies","primary_attr":"int","attack_type":"Ranged","roles":["Nuker","Disabler"],"legs":6},{"id":106,"name":"npc_dota_hero_ember_spirit","localized_name":"Ember Spirit","primary_attr":"agi","attack_type":"Melee","roles":["Carry","Escape","Nuker","Disabler","Initiator"],"legs":2},{"id":107,"name":"npc_dota_hero_earth_spirit","localized_name":"Earth Spirit","primary_attr":"str","attack_type":"Melee","roles":["Nuker","Escape","Disabler","Initiator","Durable"],"legs":2},{"id":108,"name":"npc_dota_hero_abyssal_underlord","localized_name":"Underlord","primary_attr":"str","attack_type":"Melee","roles":["Support","Nuker","Disabler","Durable","Escape"],"legs":2},{"id":109,"name":"npc_dota_hero_terrorblade","localized_name":"Terrorblade","primary_attr":"agi","attack_type":"Melee","roles":["Carry","Pusher","Nuker"],"legs":2},{"id":110,"name":"npc_dota_hero_phoenix","localized_name":"Phoenix","primary_attr":"str","attack_type":"Ranged","roles":["Support","Nuker","Initiator","Escape","Disabler"],"legs":2},{"id":111,"name":"npc_dota_hero_oracle","localized_name":"Oracle","primary_attr":"int","attack_type":"Ranged","roles":["Support","Nuker","Disabler","Escape"],"legs":2},{"id":112,"name":"npc_dota_hero_winter_wyvern","localized_name":"Winter Wyvern","primary_attr":"int","attack_type":"Ranged","roles":["Support","Disabler","Nuker"],"legs":2},{"id":113,"name":"npc_dota_hero_arc_warden","localized_name":"Arc Warden","primary_attr":"agi","attack_type":"Ranged","roles":["Carry","Escape","Nuker"],"legs":2},{"id":114,"name":"npc_dota_hero_monkey_king","localized_name":"Monkey King","primary_attr":"agi","attack_type":"Melee","roles":["Carry","Escape","Disabler","Initiator"],"legs":2},{"id":119,"name":"npc_dota_hero_dark_willow","localized_name":"Dark Willow","primary_attr":"int","attack_type":"Ranged","roles":["Support","Nuker","Disabler","Escape"],"legs":2},{"id":120,"name":"npc_dota_hero_pangolier","localized_name":"Pangolier","primary_attr":"agi","attack_type":"Melee","roles":["Carry","Nuker","Disabler","Durable","Escape","Initiator"],"legs":2}]
```

Comment:
Is this file a download? Are we sure it's correct? E.g., Crystal Maiden doesn't have the carry role.

Reply:
Yes it is downloaded from https://api.opendota.com/api/heroes. I would guess that the role information is probably equivalent to what you would see in the hero browser in the Dota 2 client. Either way I am not actually using the role information (we actually have another source for that under our control which is the !draft spreadsheet)